src/seedu/addressbook/logic/Logic.java

```diff
@@ -96,4 +98,11 @@ private void recordResult(CommandResult result) {
             lastShownList = personList.get();
         }
     }
+
+    /**
+     * @return true if parsed command mutates data, false otherwise.
+     */
+    public boolean isMutatingCommand(String inputCommand) {
+        return new Parser().parseCommand(inputCommand).isMutating();
```

Comment:
What is the use of this method when you are checking in the execute method above anyway?